bundles/org.openhab.binding.nanoleaf/src/main/java/org/openhab/binding/nanoleaf/internal/OpenAPIUtils.java

```diff
@@ -97,18 +123,21 @@ public static ContentResponse sendOpenAPIRequest(Request request)
             if (clientException.getCause() instanceof HttpResponseException) {
                 if (((HttpResponseException) clientException.getCause()).getResponse()
                         .getStatus() == HttpStatus.UNAUTHORIZED_401) {
-                    logger.warn("OpenAPI request unauthorized. Invalid authorization token.");
+                    LOGGER.warn("OpenAPI request unauthorized. Invalid authorization token.");
                     throw new NanoleafUnauthorizedException("Invalid authorization token");
                 }
             }
             throw new NanoleafException(
-                    String.format("Failed to send OpenAPI request: %s", clientException.getMessage()));
+                    String.format("Failed to send OpenAPI request: %s exception class: %s", clientException.getMessage(), clientException.getClass()));
         }
     }
 
-    public static boolean checkRequiredFirmware(String currentFirmwareVersion) {
+    public static boolean checkRequiredFirmware(String modelId, String currentFirmwareVersion) {
         int[] currentVer = getFirmwareVersionNumbers(currentFirmwareVersion);
 
+        int[] requiredVer = getFirmwareVersionNumbers(
+                modelId.equals(MODEL_ID_LIGHTPANELS) ? API_MIN_FW_VER_LIGHTPANELS : API_MIN_FW_VER_CANVAS);
```

Comment:
Position literals first in `String` comparisons - that way if the `String` is `null` you won't get a `NullPointerException`, it'll just return `false`. [source: PMD](http://pmd.sourceforge.net/pmd-4.3.0/rules/design.html#PositionLiteralsFirstInComparisons)